cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/FlowProgramRunner.java

```diff
@@ -251,6 +251,8 @@ protected void doCommand(String name, Object value) throws Exception {
         changeInstances(command.get("flowlet"), Integer.valueOf(command.get("newInstances")));
       } catch (Throwable t) {
         LOG.error(String.format("Fail to change instances: %s", command), t);
+        stop();
```

Comment:
This is to make it behave the same as in distributed mode.

I chatted with @chtyim and the distributed (Twill) flow runner stops the flow in this case, because it has come into an inconsistent state, and flowlets may consumer from (partitioned) queues that are inconsistent with the actual partitioning of the instances.